### PR TITLE
Add trigger to pyopenms nightlies test / use pyopenms-versions in pyopenms-wheels

### PR DESCRIPTION
**Pull request by matteopilz**
## Description
The trigger should work, I actually don't know how to test it.
I'm not sure about the pyopenms-wheels changes, maybe someone could take a look at it.

## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

**Comment by jpfeuffer** [comment]
Hmm no sorry but matrix is too wasteful. Why didn't you just keep the for loop and instead of checking the available pythons on the system, use the predefined versions?

**Comment by jpfeuffer** [comment]
I think you can skip the additional parsing step and just parse the file in the shell. Would be cleaner in my opinion.

**Comment by jpfeuffer** [comment]
You can use the "jq" tool. It is pre-installed on all GitHub runners. You might need to install it on Linux in this workflow, since we use the manylinux container.

**Comment by matteopilz** [comment]
Does jq need to be installed for Windows and MacOS as well?

**Comment by jpfeuffer** [comment]
No. Should be preinstalled: https://docs.github.com/de/actions/using-github-hosted-runners/about-github-hosted-runners#supported-software

File: .github/workflows/pyopenms-wheels.yml [hunk 1]
```diff
@@ -310,14 +307,8 @@ jobs:
     - name: Build on manylinux2014
       shell: bash
       run: |
+            sudo apt install -y jq
```
Comment (jpfeuffer) [inline .github/workflows/pyopenms-wheels.yml:310]: I think it is a CentOS image.
```suggestion
            sudo yum install -y jq
```

**Review by jpfeuffer — APPROVED** [review]
Yup. We can try it like that.

**Comment by jpfeuffer** [comment]
I just saw that even if the pyopenms-wheels workflows was run for a release branch, the "test" action will test the nightlies.

**Comment by jpfeuffer** [comment]
pyopenms-nightly will also not work after @timosachsenberg s latest changes

**Comment by timosachsenberg** [comment]
change "pip install --index-url https://pypi.cs.uni-tuebingen.de/simple/ pyopenms-nightly "
to
"pip install --index-url https://pypi.cs.uni-tuebingen.de/simple/ pyopenms"

**Comment by jpfeuffer** [comment]
This broke our builds on both win and lnx.

https://github.com/OpenMS/OpenMS/actions/runs/4379918068/jobs/7666383333

**Comment by jpfeuffer** [comment]
@matteopilz

**Pull request merged**